tyk-pro/templates/deployment-dash.yaml

```diff
@@ -71,9 +71,9 @@ spec:
           - name: TYK_DB_TYKAPI_PORT
             value: "{{ .Values.gateway.service.port }}"
           {{- if .Values.gateway.control.enabled }}
-          - name: TYK_DB_TYKAPI_HOST
+          - name: TYK_GW_CONTROLAPIHOSTNAME
             value: "{{ include "tyk-pro.gwproto" . }}://gateway-control-svc-{{ include "tyk-pro.fullname" . }}.{{ .Release.Namespace }}"
-          - name: TYK_DB_TYKAPI_PORT
+          - name: TYK_GW_CONTROLAPIPORT
```

Comment:
environment variables are wrong based on the [docs](https://tyk.io/docs/tyk-oss-gateway/configuration/#control_api_hostname). Smoke tests were failing during upgrade due to this error.

Reply:
@abhishiekgowtham This spec is for the dashboard, not the gateway. `TYK_GW_` prefix is no-op on the dashboard, so these variables will do nothing here.

The previous variables are correct in this context. What seems to be wrong is that the condition block is incorrect — if `gateway.control.enabled` is `true` then the variables will be specified twice and the first one will likely win out.

Reply:
@excieve Addressed. But having the same environment variables enabled [fails](https://github.com/TykTechnologies/tyk-helm-chart/pull/122/checks?check_run_id=2681563569#step:6:16) during an helm upgrade. The dashboard should speak to control api when `gateway.control.enabled` is enabled, if not connect to the default `hostname` and `port`.

Reply:
@abhishiekgowtham Correct, so there's a block that duplicates these env vars, which was likely causing issues. I see you corrected it now with an `else` clause and the tests are passing. So this seems to be fixed.